node/service/src/overseer.rs

```diff
@@ -161,41 +170,45 @@ where
 	Spawner: 'static + SpawnNamed + Clone + Unpin,
 {
 	use polkadot_node_subsystem_util::metrics::Metrics;
+	use std::iter::FromIterator;
+
+	let metrics: polkadot_overseer::Metrics =
+		<polkadot_overseer::Metrics as MetricsTrait>::register(registry)?;
 
-	let all_subsystems = AllSubsystems {
-		availability_distribution: AvailabilityDistributionSubsystem::new(
+	let builder = Overseer::builder()
+		.availability_distribution(AvailabilityDistributionSubsystem::new(
 			keystore.clone(),
 			IncomingRequestReceivers { pov_req_receiver, chunk_req_receiver },
 			Metrics::register(registry)?,
-		),
-		availability_recovery: AvailabilityRecoverySubsystem::with_chunks_only(
+		))
+		.availability_recovery(AvailabilityRecoverySubsystem::with_chunks_only(
 			available_data_req_receiver,
 			Metrics::register(registry)?,
-		),
-		availability_store: AvailabilityStoreSubsystem::new(
+		))
+		.availability_store(AvailabilityStoreSubsystem::new(
 			parachains_db.clone(),
 			availability_config,
 			Metrics::register(registry)?,
-		),
-		bitfield_distribution: BitfieldDistributionSubsystem::new(Metrics::register(registry)?),
-		bitfield_signing: BitfieldSigningSubsystem::new(
+		))
+		.bitfield_distribution(BitfieldDistributionSubsystem::new(Metrics::register(registry)?))
+		.bitfield_signing(BitfieldSigningSubsystem::new(
 			spawner.clone(),
 			keystore.clone(),
 			Metrics::register(registry)?,
-		),
-		candidate_backing: CandidateBackingSubsystem::new(
+		))
+		.candidate_backing(CandidateBackingSubsystem::new(
 			spawner.clone(),
 			keystore.clone(),
 			Metrics::register(registry)?,
-		),
-		candidate_validation: CandidateValidationSubsystem::with_config(
```

Comment:
Looks like it is going to be easier to forget to add a subsystem now?

Reply:
Tracked in #3772 it will currently fail to run.

Reply:
Got it - immediate panic on startup with precise error is already pretty good. I would leave it at that, if the alternative is more macro/meta programming.